### PR TITLE
Update ci to use ros2 for building instead of ros 601

### PR DESCRIPTION
**Pull request by OZiad**
# Assignee Section

## Description
Update ROS 2 installation script to verify and install ROS Humble and change CI to use old Arduino script

### CODE REVIEW

File: scripts/ros_install.sh
Comment (steamypotato) [inline scripts/ros_install.sh:1]: EnvironmentSetup.sh references 'install_ros.sh' (line 21), but file is called is 'ros_install.sh'

File: EnvironmentSetup.sh
Comment (steamypotato) [inline EnvironmentSetup.sh:1]: line 39 (rosdep install -i --from-path src --rosdistro foxy -y) should be updated for ROS humble

**Review by steamypotato — CHANGES_REQUESTED** [review]

File: scripts/ros_install.sh [hunk 1]
```diff
@@ -7,19 +7,33 @@ sudo update-locale LC_ALL=en_CA.UTF-8 LANG=en_CA.UTF-8
...
 sudo apt -qq install curl
-echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
+
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list >/dev/null
 
 echo "Updating sources and upgrading system packages..."
 sudo apt -qq update
-sudo apt -qq upgrade -y
+sudo apt -qq upgrade
```
Comment (VimFreeman) [inline scripts/ros_install.sh:22]: the -y is necessary or else script will stall until user confirms the upgrade

**Comment by VimFreeman** [comment]
image

**Comment by VimFreeman** [comment]
@OZiad did you try to run EnvironmentSetup.sh in a new user?

**Comment by OZiad** [comment]
> @OZiad did you try to run EnvironmentSetup.sh in a new user?

Yea but I'm not able to replicate the CMake error, it seems to run fine on the test user I made after fixing the ros dependency issue

**Pull request merged**